### PR TITLE
2774 add futures executor threadpool

### LINKED ISSUE #2774: examples: Add a `futures::executor::ThreadPool` example
We should include an example in `examples` called `custom-executor` that will show how to use tokio io/timers within a custom executor.

This [gist](https://gist.github.com/LucioFranco/ec8c6617b7193fd062d1ddc32b5bb991) is a good starting point. We likely want to show a couple things:

1) Setting up a tokio runtime that runs on a single thread and is driven without the need to explicitly spawn futures onto it.
2) Show how to "enter" the tokio context on each poll invocation of the custom executor.

### PR DESCRIPTION
**Pull request by carlosb1**
closes #2774

### CODE REVIEW

File: examples/custom-executor.rs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            self.inner.spawn_ok(TokioIo { handle, inner: f });
+        }
+    }
+
+    #[pin_project]
+    struct TokioIo<F> {
+        #[pin]
+        inner: F,
+        handle: Handle,
+    }
```
Comment (Darksonn) [inline examples/custom-executor.rs:68]: With the new `TokioContext` constructor, you should no longer need this type.

Comment (carlosb1) [reply examples/custom-executor.rs]: Done!, It should be correct now..

File: examples/custom-executor.rs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    struct ThreadPool {
+        inner: futures::executor::ThreadPool,
+        rt: tokio::runtime::Runtime,
+    }
+
+    static EXECUTOR: Lazy<ThreadPool> = Lazy::new(|| {
+        // Spawn tokio runtime on a single background thread
+        // enabling IO and timers.
+        let rt = tokio::runtime::Builder::new_current_thread()
```
Comment (Darksonn) [inline examples/custom-executor.rs:41]: This needs to be a `new_multi_thread` with one worker thread if we want a background thread.

File: examples/custom-executor.rs [hunk 3]
```diff
@@ -0,0 +1,57 @@
+// This example shows how to use the tokio runtime with any other executor
+//
+// The main components are a spawn fn that will wrap futures in a special future
+// that will always enter the tokio context on poll. This only spawns on extra thread
```
Comment (Darksonn) [inline examples/custom-executor.rs:4]:
```suggestion
// that will always enter the tokio context on poll. This only spawns one extra thread
```

File: examples/custom-executor.rs (hunk 2 above)
Comment (carlosb1) [reply examples/custom-executor.rs]: Done!, sorry I didn't  see it

**Review by Darksonn — APPROVED** [review]

File: examples/Cargo.toml [hunk 4]
```diff
@@ -15,13 +15,16 @@ async-stream = "0.3"
...
-futures = "0.3.0"
+futures = { version = "0.3.0", features = ["thread-pool"]}
 http = "0.2"
 serde = "1.0"
 serde_derive = "1.0"
 serde_json = "1.0"
 httparse = "1.0"
 time = "0.1"
+once_cell = "1.5.2"
+pin-project = "1.0.1"
```
Comment (Darksonn) [inline examples/Cargo.toml:26]: This dependency is no longer needed.

**Pull request merged**